src/main/java/frc/utils/math/MirrorMath.java

```diff
@@ -18,4 +21,25 @@ public static double getMirroredY(double y) {
 		return Field.WIDTH_METERS - y;
 	}
 
+	public static Pose2d getMirroredPose(Pose2d pose2d, boolean invertY, boolean invertAngle) {
+		if (invertY) {
+			return invertAngle
+				? new Pose2d(getMirroredX(pose2d.getX()), getMirroredY(pose2d.getY()), pose2d.getRotation())
+				: new Pose2d(getMirroredX(pose2d.getX()), getMirroredY(pose2d.getY()), getMirroredAngle(pose2d.getRotation()));
+		}
+		return invertAngle
+			? new Pose2d(getMirroredX(pose2d.getX()), pose2d.getY(), pose2d.getRotation())
+			: new Pose2d(getMirroredX(pose2d.getX()), pose2d.getY(), getMirroredAngle(pose2d.getRotation()));
+	}
+
+	public static Translation3d getMirroredTranslation3d(Translation3d translation3d, boolean invertZ) {
+		return invertZ
+			? new Translation3d(getMirroredX(translation3d.getX()), getMirroredY(translation3d.getY()), 0 - translation3d.getZ())
+			: new Translation3d(getMirroredX(translation3d.getX()), getMirroredY(translation3d.getY()), translation3d.getZ());
+	}
+
+	public static Rotation3d getMirroredRotation3d(Rotation3d rotation3d) {
```

Comment:
```suggestion
	public static Rotation3d getMirroredRotation3d(Rotation3d rotation) {
```
no need to specify the type